onnx/shape_inference/implementation.h

```diff
@@ -86,22 +86,25 @@ struct GraphInferenceContext {
           outer_scope_value_types_by_name_in,
       const std::unordered_map<std::string, int> opset_imports_in,
       SymbolTable* symbol_table_in = nullptr,
+      const ModelLocalFunctionsMap& model_local_functions_in = {},
       const ISchemaRegistry* schema_registry_in = OpSchemaRegistry::Instance(),
-      const int ir_version_in = IR_VERSION,
-      const ModelLocalFunctionsMap& model_local_functions_in = {})
+      std::unordered_map<std::string, TensorShapeProto>* generated_shape_data_by_name_in = nullptr,
+      const int ir_version_in = IR_VERSION)
       : outer_scope_value_types_by_name{&outer_scope_value_types_by_name_in},
         opset_imports{opset_imports_in},
         symbol_table{symbol_table_in},
+        model_local_functions{model_local_functions_in},
         schema_registry{schema_registry_in},
-        ir_version{ir_version_in},
-        model_local_functions{model_local_functions_in} {}
+        generated_shape_data_by_name{generated_shape_data_by_name_in},
+        ir_version{ir_version_in} {}
 
   const std::unordered_map<std::string, TypeProto*>* outer_scope_value_types_by_name;
   const std::unordered_map<std::string, int> opset_imports;
   SymbolTable* symbol_table;
+  const ModelLocalFunctionsMap& model_local_functions;
   const ISchemaRegistry* schema_registry;
+  std::unordered_map<std::string, TensorShapeProto>* generated_shape_data_by_name;
   const int ir_version;
-  const ModelLocalFunctionsMap& model_local_functions;
```

Comment:
FYI: I changed the order here for `ModelLocalFunctionsMap& model_local_functions` to make it consistent to other APIs.